tests/validation/tests/v3_api/common.py

```diff
@@ -2865,3 +2871,67 @@ def validate_cluster_role_rbac(cluster_role, command, authorization, name):
             "{} should NOT have the authorization to run {}".format(
                 cluster_role, command)
 
+
+def wait_until_app_v2_deployed(client, app_name, timeout=DEFAULT_APP_V2_TIMEOUT):
+    """
+    List all installed apps and check for the state of "app_name" to see
+    if it == "deployed"
+    :param client: cluster client for the user
+    :param app_name: app which is being installed
+    :param timeout: time for the app to come to Deployed state
+    :return:
+    """
+    start = time.time()
+    app = client.list_catalog_cattle_io_app()
+    while True:
+        app_list = []
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out waiting for state to get to Deployed")
+        time.sleep(.5)
+        for app in app["data"]:
+            app_list.append(app["metadata"]["name"])
+            if app["metadata"]["name"] == app_name:
+                if app["status"]["summary"]["state"] == "deployed":
+                    return app_list
+        app = client.list_catalog_cattle_io_app()
+    return
+
+
+def wait_until_app_v2_uninstall(client, app_name, timeout=DEFAULT_APP_V2_TIMEOUT):
+    """
+    list all installed apps. search for "app_name" in the list
+    if app_name is NOT in list, indicates the app has been uninstalled successfully
+    :param client: cluster client for the user
+    :param app_name: app which is being unstalled
+    :param timeout: time for app to be uninstalled
+    """
+    start = time.time()
+    app = client.list_catalog_cattle_io_app()
+    while True:
+        app_list = []
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out waiting for state to get to Uninstalled")
+        time.sleep(.5)
+        for app in app["data"]:
+            app_list.append(app["metadata"]["name"])
+        if app_name not in app_list:
+            return app_list
+        app = client.list_catalog_cattle_io_app()
+    return
+
+
+def check_if_v2_app_installed(client, chart_name):
```

Comment:
1. the name of the method does not imply what it actually does: uninstall the app.
2. Not all apps have the crd part. So instead of checking both the app and app_crd, we can just call the method twice with different arguments( app, app_crd). This makes the method more reusable.

Reply:
done